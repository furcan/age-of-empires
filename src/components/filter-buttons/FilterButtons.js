import { FiCheck } from 'react-icons/fi';

import './FilterButtons.scss';

function FilterButtons(props) {
  const { onClickHandler, classListAge } = props;

  return (
    <div className="buttons-wrapper">
      <button type="button" name="All" onClick={onClickHandler} className={classListAge.All ? 'selected' : ''}>
        {classListAge.All && <FiCheck />}
        <span>All</span>
      </button>
      <button type="button" name="Dark" onClick={onClickHandler} className={classListAge.Dark ? 'selected' : ''}>
        {classListAge.Dark && <FiCheck />}
        <span>Dark</span>
      </button>
      <button type="button" name="Feudal" onClick={onClickHandler} className={classListAge.Feudal ? 'selected' : ''}>
        {classListAge.Feudal && <FiCheck />}
        <span>Feudal</span>
      </button>
      <button type="button" name="Castle" onClick={onClickHandler} className={classListAge.Castle ? 'selected' : ''}>
        {classListAge.Castle && <FiCheck />}
        <span>Castle</span>
      </button>
      <button type="button" name="Imperial" onClick={onClickHandler} className={classListAge.Imperial ? 'selected' : ''}>
        {classListAge.Imperial && <FiCheck />}
        <span>Imperial</span>
      </button>
    </div>
  );
}

export default FilterButtons;
