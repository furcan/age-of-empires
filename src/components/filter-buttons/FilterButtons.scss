@import "../../styles/variables";

.buttons-wrapper {
  max-width: 100%;
  display: block;
  white-space: nowrap;
  overflow: auto hidden;
  padding: 8px 0;

  button {
    position: relative;
    cursor: pointer;
    transition: all 0.25s ease-in-out;
    padding: 8px 16px;
    margin: 0 12px 8px 0;
    border: none;
    outline: none;
    box-shadow: 0 0 16px -8px rgba($color: black, $alpha: 0.3);
    background: $light;
    color: $dark;
    border-radius: 8px;
    display: inline-flex;
    flex-wrap: wrap;
    align-items: center;
    font-weight: 600;
    font-size: 14px;

    @media (min-width: $breakpoint-tablet) {
      font-size: 16px;
    }

    * {
      pointer-events: none;
    }

    i,
    svg {
      margin-right: 10px;
      transform: scale(1.1);
      transform-origin: center center;
    }

    &:hover {
      background: $gray;
      color: $light;
    }

    &.selected {
      background: $dark;
      color: $light;
    }
  }
}
