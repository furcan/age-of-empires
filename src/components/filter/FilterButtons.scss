@import "../../styles/variables";

.buttons-wrapper {
  max-width: 100%;
  display: block;
  white-space: nowrap;
  overflow: auto hidden;
  padding: 8px 0;

  button {
    position: relative;
    cursor: pointer;
    transition: all 0.25s ease-in-out;
    padding: 10px 18px;
    margin: 0 12px 8px 0;
    border: none;
    outline: none;
    box-shadow: 0 0 16px -8px rgba($color: black, $alpha: 0.3);
    background: $white;
    color: $dark;
    border-radius: 8px;
    display: inline-flex;
    flex-wrap: wrap;
    align-items: center;
    font-weight: 600;
    font-size: 14px;

    @media (min-width: $breakpoint-tablet) {
      font-size: 16px;
    }

    * {
      pointer-events: none;
    }

    i,
    svg {
      margin-right: 10px;
      transform: scale(1.1);
      transform-origin: center center;
    }

    &:hover {
      background: $gray;
      color: $white;
    }

    &.selected {
      background: $dark;
      color: $white;
    }
  }
}
