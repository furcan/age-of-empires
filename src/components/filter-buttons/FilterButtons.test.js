
import renderer from 'react-test-renderer';

import FilterButtons from './FilterButtons';

it('The FilterButtons component has been rendered as expected.', () => {
  const tree = renderer.create(<FilterButtons />).toJSON();
  expect(tree).toMatchSnapshot();
});
