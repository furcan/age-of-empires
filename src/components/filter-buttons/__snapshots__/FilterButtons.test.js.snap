// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The FilterButtons component has been rendered as expected. 1`] = `
<div
  className="buttons-wrapper"
>
  <button
    className=""
    name="All"
    type="button"
  >
    <span>
      All
    </span>
  </button>
  <button
    className=""
    name="Dark"
    type="button"
  >
    <span>
      Dark
    </span>
  </button>
  <button
    className=""
    name="Feudal"
    type="button"
  >
    <span>
      Feudal
    </span>
  </button>
  <button
    className=""
    name="Castle"
    type="button"
  >
    <span>
      Castle
    </span>
  </button>
  <button
    className=""
    name="Imperial"
    type="button"
  >
    <span>
      Imperial
    </span>
  </button>
</div>
`;
